### PR TITLE
Remove events deprecated field

### PR DESCRIPTION
**Pull request by wojtek-t**
```release-note
The unused `series.state` field, deprecated since v1.14, is removed from the `events.k8s.io/v1beta1` and `v1` Event types.
```

/kind cleanup

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -5215,19 +5215,8 @@ type EventSeries struct {
 	Count int32 `json:"count,omitempty" protobuf:"varint,1,name=count"`
 	// Time of the last occurrence observed
 	LastObservedTime metav1.MicroTime `json:"lastObservedTime,omitempty" protobuf:"bytes,2,name=lastObservedTime"`
-	// State of this Series: Ongoing or Finished
-	// Deprecated. Planned removal for 1.18
-	State EventSeriesState `json:"state,omitempty" protobuf:"bytes,3,name=state"`
```
Comment (wojtek-t) [inline staging/src/k8s.io/api/core/v1/types.go:5220]: @liggitt - the test failures are relevant to this change. Does it mean we can't remove those fields?
Or should I fix the tests?

Comment (liggitt) [reply staging/src/k8s.io/api/core/v1/types.go:5220]: per https://github.com/kubernetes/kubernetes/pull/75987#issuecomment-479309114, go ahead and update the tests (run with `UPDATE_COMPATIBILITY_FIXTURE_DATA=true`) and capture the test output to include as a diff

Comment (liggitt) [inline staging/src/k8s.io/api/core/v1/types.go:5220]: tombstone the field so we don't reuse this proto or json tag. examples:

https://github.com/kubernetes/kubernetes/blob/e97c570a4ba5ba1e2285d3278396937feaa15385/staging/src/k8s.io/api/core/v1/types.go#L4311-L4313

https://github.com/kubernetes/kubernetes/blob/e97c570a4ba5ba1e2285d3278396937feaa15385/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1118-L1119

File: staging/src/k8s.io/api/testdata/v1.18.0/events.k8s.io.v1beta1.Event.yaml [hunk 2]
```diff
@@ -60,5 +60,4 @@ reportingInstance: "20"
 series:
   count: -1971381490
   lastObservedTime: "2429-02-18T17:57:56.343118Z"
-  state: 鯹)晿<
```
Comment (liggitt) [inline staging/src/k8s.io/api/testdata/v1.18.0/events.k8s.io.v1beta1.Event.yaml:63]: actually commit the .after_roundtrip files, don't modify the originals...

File: staging/src/k8s.io/api/core/v1/types.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/api/core/v1/types.go:5220]: Thanks for catching.

File: staging/src/k8s.io/api/testdata/v1.18.0/events.k8s.io.v1beta1.Event.yaml (hunk 2 above)
Comment (wojtek-t) [reply staging/src/k8s.io/api/testdata/v1.18.0/events.k8s.io.v1beta1.Event.yaml]: Done - sorry, didn't realize we should commit those.

**Pull request merged**